### PR TITLE
Update models with nullability specifiers

### PR DESCRIPTION
**Pull request by danzimm**
Other things done include properly differentiate account and user photo models and update color transformer

### CODE REVIEW

File: CocoaBloc/Models/SBUserPhoto.m [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+			 @"smallURL"		: @"images.small_url",
+			 @"mediumURL"		: @"images.medium_url",
+			 @"largeURL"		: @"images.large_url",
+			 @"thumbnailURL"	: @"images.thumbnail_url",
+			 @"kind"			: @"kind"};
+}
+
+- (BOOL)isEqual:(id)object {
+	__block BOOL equal = YES;
+	NSArray *properties = [[[self class] JSONKeyPathsByPropertyKey] allKeys];
```
Comment (JRHeaton) [inline CocoaBloc/Models/SBUserPhoto.m]: curious what this is achieving over the standard isEqual

Comment (danzimm) [inline CocoaBloc/Models/SBUserPhoto.m]: I copied it from `SBObject` - this doesn't inherit from `SBObject` because there is no identifier

Comment (danzimm) [inline CocoaBloc/Models/SBUserPhoto.m]: In other words no idea

Comment (JRHeaton) [inline CocoaBloc/Models/SBUserPhoto.m]: I think that the implementation left in `SBObject` may have been an old implementation that slipped in. Perhaps we can write a one-liner unit test or at least maybe verify that it works w/ [Mantle's own provided `-[MTLModel isEqual:]`](https://github.com/Mantle/Mantle/blob/master/Mantle/MTLModel.m#L319-L332)?

File: CocoaBloc/Models/SBFanClub.h [hunk 2]
```diff
@@ -10,27 +10,34 @@
...
 
+@property (nonatomic) NSNumber *canPostBlogs; // Bool
+@property (nonatomic) NSNumber *canPostPhotos; // Bool
+@property (nonatomic) NSNumber *canPostStatuses; // Bool
+@property (nonatomic) NSNumber *canPostAudio; // Bool
+@property (nonatomic) NSNumber *canPostVideos; // Bool
+
+// TODO(danzimm): PHP has this as present for legacy reasons. Switch over to using `user_membership`
+@property (nonatomic, nullable) NSNumber *userTier;
+
```
Comment (JRHeaton) [inline CocoaBloc/Models/SBFanClub.h]: Added [this issue](https://github.com/stagebloc/CocoaBloc/issues/119) so we can pull this TODO. Felt that this particular one was important enough to warrant it.

Comment (danzimm) [inline CocoaBloc/Models/SBFanClub.h]: Removed comment

File: CocoaBloc/Models/SBUserPhoto.m (hunk 1 above)
Comment (danzimm) [inline CocoaBloc/Models/SBUserPhoto.m]: Added unit test, unit test passes with `-[MTLModel isEqual:]` so keeping just that

**Pull request merged**